src/lib/coda_commands/coda_commands.ml

```diff
@@ -191,6 +191,121 @@ let setup_user_command ~fee ~nonce ~valid_until ~memo ~sender_kp
   let signed_user_command = User_command.sign sender_kp payload in
   User_command.forget_check signed_user_command
 
+module Hardware_wallet = struct
+  open Snark_params
+
+  let hardware_wallet_script = "cli/sign.py"
+
+  let decode_field : string -> Snark_params.Tick.Field.t =
+   fun field ->
+    Bytes.of_string field
+    |> B58.decode Base58_check.coda_alphabet
+    |> Bytes.to_string
+    |> String.foldi ~init:Bigint.zero ~f:(fun i acc byte ->
+           Bigint.(acc lor (of_int (Char.to_int byte) lsl Int.( * ) 8 i)) )
+    |> Tick.Bigint.of_bignum_bigint |> Tick.Bigint.to_field
+
+  let decode_scalar : string -> Snark_params.Tock.Field.t =
+   fun scalar ->
+    Bytes.of_string scalar
+    |> B58.decode Base58_check.coda_alphabet
+    |> Bytes.to_string
+    |> String.foldi ~init:Bigint.zero ~f:(fun i acc byte ->
+           Bigint.(acc lor (of_int (Char.to_int byte) lsl Int.( * ) 8 i)) )
```

Comment:
`decode_field` and `decode_scalar` appear to be identical except for the last lines and the name of the bound variable. Could you factor that into a helper function?